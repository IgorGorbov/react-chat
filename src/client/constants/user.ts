export enum Status {
  Online = 'Online',
  Offline = 'Offline',
  Away = 'Away',
  Busy = 'Busy',
}

export const statusColors = {
  [Status.Online]: 'green',
  [Status.Offline]: 'grey',
  [Status.Away]: '#dede0f',
  [Status.Busy]: 'red',
};
