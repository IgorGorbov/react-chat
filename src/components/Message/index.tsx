import * as React from 'react';
import ReactWOW from 'react-wow';
import {
  UserAvatar,
  TextMessage,
  MessageContainer,
  MessageInfoContainer,
  MessageArrow,
  MessageDate,
} from '../../styledComponents';

interface IMessageProps {
  key: number;
  message: IMessage;
  currentUser: IUser;
  selectedMessages: number[];
  selectMessage: (messageId: number) => { type: string; payload: number };
  className: string;
}

const Message = (props: IMessageProps): JSX.Element | null => {
  const { currentUser, message, selectMessage, selectedMessages } = props;
  const isMessageTo = currentUser.id === message.from.id ? 'isMessageTo' : '';
  const isSelected = selectedMessages.includes(message.id) ? ' isSelected' : '';
  if (message.deletedBy.includes(currentUser.id)) {
    return null;
  }
  return (
    <ReactWOW animation={isMessageTo ? 'fadeInRight' : 'fadeInLeft'}>
      <MessageContainer
        key={message.id}
        className={isMessageTo}
        onClick={() => selectMessage(message.id)}
      >
        <UserAvatar
          maxwidth="45px"
          maxheight="45px"
          bordersize="1px"
          src={message.from.avatar}
          size="tiny"
          circular={true}
        />
        <MessageInfoContainer className={isMessageTo + isSelected}>
          <MessageArrow className={isMessageTo + isSelected} />
          <TextMessage>{message.text}</TextMessage>
          <MessageDate>{message.date}</MessageDate>
        </MessageInfoContainer>
      </MessageContainer>
    </ReactWOW>
  );
};

export default Message;
