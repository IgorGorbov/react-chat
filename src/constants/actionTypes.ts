export const USER_LOGIN = 'USER_LOGIN';
export const CHANGE_USER_STATUS = 'CHANGE_USER_STATUS';
export const FETCH_ALL_USERS = 'FETCH_ALL_USERS';
export const SEARCH_CHAT = 'SEARCH_CHAT';
export const CHANGE_CHAT = 'CHANGE_CHAT';
export const NEW_CHAT = 'NEW_CHAT';
export const DELETE_CHAT = 'DELETE_CHAT';
export const NEW_MESSAGE = 'NEW_MESSAGE';
export const CHOOSE_USER = 'CHOOSE_USER';
export const SELECT_MESSAGE = 'SELECT_MESSAGE';
export const DELETE_MESSAGES = 'DELETE_MESSAGES';
export const READ_MESSAGES = 'READ_MESSAGES';
