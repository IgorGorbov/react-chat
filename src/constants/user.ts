export const defaultAvatar = 'https://emilcarlsson.se/assets/mikeross.png';
export enum Status {
  Online = "Online",
  Offline = "Offline",
  Away = "Away",
  Busy = "Busy"
}

export const statusColors = {
  [Status.Online]: 'green',
  [Status.Offline]: 'grey',
  [Status.Away]: 'yellow',
  [Status.Busy]: 'red'
};
