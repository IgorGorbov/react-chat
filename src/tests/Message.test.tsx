import * as React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import Message from '../client/components/Message/index';
import {
  MessageContainer,
  UserAvatar,
  MessageInfoContainer,
  MessageArrow,
  TextMessage,
  MessageDate,
} from '../client/styledComponents/index';

import { Status } from '../client/constants/user';

describe('testing Message', () => {
  interface IMessageProps {
    key: number;
    message: IMessage;
    currentUser: IUser;
    selectedMessages: number[];
    selectMessage: (messageId: number) => { type: string; payload: number };
    className: string;
  }
  const FakeSelectMessage = jest.fn();
  const initState: IMessageProps = {
    key: 1,
    message: {
      id: 1,
      from: {
        id: 1,
        name: 'test',
        avatar: 'test',
        status: Status.Online,
      },
      to: {
        id: 1,
        name: 'test',
        avatar: 'test',
        status: Status.Online,
      },
      text: 'Text message',
      date: '16.09.2018',
      isRead: false,
      deletedBy: [],
    },
    currentUser: {
      id: 1,
      name: 'test',
      avatar: 'test',
      status: Status.Online,
    },
    selectMessage: FakeSelectMessage,
    selectedMessages: [],
    className: '',
  };

  let wrapper: any;

  beforeAll(() => {
    wrapper = shallow(<Message {...initState} />);
  });

  it('should contain MessageContainer', () => {
    const messageContainer = wrapper.find(MessageContainer);
    expect(messageContainer).toHaveLength(1);

    messageContainer.simulate('click');
    expect(FakeSelectMessage).toHaveBeenCalled();
  });

  it('should contain UserAvatar', () => {
    expect(wrapper.find(UserAvatar)).toHaveLength(1);
  });

  it('should contain MessageInfoContainer', () => {
    const messageInfoContainer = wrapper.find(MessageInfoContainer);
    expect(messageInfoContainer).toHaveLength(1);
  });

  it('should contain MessageArrow', () => {
    expect(wrapper.find(MessageArrow)).toHaveLength(1);
  });

  it('should contain TextMessage', () => {
    const textMessage = wrapper.find(TextMessage);
    expect(textMessage).toHaveLength(1);
    expect(textMessage.dive().text()).toEqual(initState.message.text);
  });

  it('should contain MessageDate', () => {
    const messageDate = wrapper.find(MessageDate);
    expect(messageDate).toHaveLength(1);
    expect(messageDate.dive().text()).toEqual(initState.message.date);
  });

  it('snapshot Message', () => {
    const tree = toJSON(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
