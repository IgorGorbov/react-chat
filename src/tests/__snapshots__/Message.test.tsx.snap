// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Message snapshot Message 1`] = `
<Message
  className=""
  currentUser={
    Object {
      "avatar": "test",
      "id": 1,
      "name": "test",
      "status": "Online",
    }
  }
  message={
    Object {
      "date": "16.09.2018",
      "deletedBy": Array [],
      "from": Object {
        "avatar": "test",
        "id": 1,
        "name": "test",
        "status": "Online",
      },
      "id": 1,
      "isRead": false,
      "text": "Text message",
      "to": Object {
        "avatar": "test",
        "id": 1,
        "name": "test",
        "status": "Online",
      },
    }
  }
  selectMessage={
    [MockFunction] {
      "calls": Array [
        Array [
          1,
        ],
      ],
    }
  }
  selectedMessages={Array []}
>
  <ReactWOW
    animateClass="animated"
    animation="fadeInRight"
    callback={[Function]}
    delay=""
    disabled={false}
    duration=""
    iteration=""
    offset={0}
    overflow={false}
    resize={true}
    scroll={true}
  >
    <styled.div
      className="isMessageTo fadeInRight"
      key=".$1"
      onAnimationEnd={[Function]}
      onClick={[Function]}
      style={
        Object {
          "animationName": "none",
          "visibility": "hidden",
        }
      }
    >
      <div
        className="isMessageTo fadeInRight sc-gqjmRU fEkjPM"
        onAnimationEnd={[Function]}
        onClick={[Function]}
        style={
          Object {
            "animationName": "none",
            "visibility": "hidden",
          }
        }
      >
        <Styled(Image)
          bordersize="1px"
          circular={true}
          maxheight="45px"
          maxwidth="45px"
          size="tiny"
          src="test"
        >
          <Image
            as="img"
            bordersize="1px"
            circular={true}
            className="sc-ifAKCX JhXmE"
            maxheight="45px"
            maxwidth="45px"
            size="tiny"
            src="test"
            ui={true}
          >
            <img
              bordersize="1px"
              className="ui tiny circular image sc-ifAKCX JhXmE"
              maxheight="45px"
              maxwidth="45px"
              src="test"
            />
          </Image>
        </Styled(Image)>
        <styled.div
          className="isMessageTo"
        >
          <div
            className="isMessageTo sc-VigVT iTxXta"
          >
            <styled.span
              className="isMessageTo"
            >
              <span
                className="isMessageTo sc-jTzLTM jajjDS"
              />
            </styled.span>
            <styled.p>
              <p
                className="sc-fjdhpX gAKYZH"
              >
                Text message
              </p>
            </styled.p>
            <styled.span>
              <span
                className="sc-jzJRlG dEnjoI"
              >
                16.09.2018
              </span>
            </styled.span>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </ReactWOW>
</Message>
`;
